intera_interface/src/intera_interface/limb.py

```diff
@@ -636,12 +639,8 @@ def ik_request(self, pose,
             return False
         limb_joints = {}
         # Check if result valid, and type of seed ultimately used to get solution
-        if (resp.result_type[0] > 0):
-            seed_str = {
-                        ikreq.SEED_USER: 'User Provided Seed',
-                        ikreq.SEED_CURRENT: 'Current Joint Angles',
-                        ikreq.SEED_NS_MAP: 'Nullspace Setpoints',
-                       }.get(resp.result_type[0], 'None')
+        if (resp.result_type[0] > 0
+                or (allow_collision and resp.result_type[0] == resp.IK_IN_COLLISION)):
```

Comment:
That truth table looks correct, but A and C are not independent. Two things to note:
- If the (result_type > 0), the solution is already checked to be collision free.
- If (result_type == IK_IN_COLLISION), there is a valid IK solution, but it's in collision.

For the line which is labeled "fishy", here is the correct interpolation:
False || (True && True) = An IK solution was found; it's in collision, but we don't care.